flash_examples/image_classification_meta_learning.py

```diff
@@ -0,0 +1,50 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import flash
+from flash.core.data.utils import download_data
+from flash.image import ImageClassificationData, ImageClassifier
+
+# 1. Create the DataModule
+download_data("https://pl-flash-data.s3.amazonaws.com/hymenoptera_data.zip", "./data")
+
+datamodule = ImageClassificationData.from_folders(
+    train_folder="data/hymenoptera_data/train/",
+    val_folder="data/hymenoptera_data/val/",
+)
+
+# 2. Build the task
+model = ImageClassifier(
+    datamodule.num_classes,
```

Comment:
Apparently, here the logic of specifying the number of classes beforehand breaks.

In particular, in the case of `training_strategy="prototypicalnetworks"` (and in general in meta-learning applications) the number of classes is only known via the `ways` in training_kwargs and can even change between training, validation, testing, and prediction. Passing a `None` currently creates a `TypeError`.